### PR TITLE
fix: add s2s auth for atracker eventstream target

### PR DESCRIPTION
**Pull request by iamar7**
### Description

Issue: https://github.com/terraform-ibm-modules/terraform-ibm-observability-instances/issues/581

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by iamar7** [comment]
`TestRunUpgradeExample` is failing due to changes in the authentication method for atracker eventstream target.
>         	Test:       	TestRunUpgradeExample
>         	Messages:   	Resource(s) identified to be updated 
>         	            	Name: atracker_eventstreams_targets
>         	            	Address: module.observability_instances.module.activity_tracker.ibm_atracker_target.atracker_eventstreams_targets["obs-upg-atdyig-es-target"]
>         	            	Actions: [update]
>         	            	DIFF:
>         	            	  Before: 
>         	            		{"eventstreams_endpoint":"SECURE_VALUE_HIDDEN_HASH:-781375d9817072fbeee234016e2762b5026acb96b517be0d9e8a5642"}
>         	            	  After: 
>         	            		{"eventstreams_endpoint":"SECURE_VALUE_HIDDEN_HASH:-a969af520df56a3578ba4a6cc31c219efebc569df249d90af2c531f1"}
>         	            	
>         	            	Change Detail:
>         	            	{
>         	            	  "actions": [
>         	            	    "update"
>         	            	  ],
>         	            	  "after": {
>         	            	    "api_version": 2,
>         	            	    "cloudlogs_endpoint": "SECURE_VALUE_HIDDEN_HASH:-569d271dfe8e452f60a74f605b790abf16872293c0e036cbe4910cdd",
>         	            	    "cos_endpoint": "SECURE_VALUE_HIDDEN_HASH:-3923b08d7b052eb18b5b4f5633bdd286703ea49faaf47c2cd6f3f590",
>         	            	    "cos_write_status": "SECURE_VALUE_HIDDEN_HASH:-7a5b112f990d9ba0c57dfae4a9b96143ca1f5693520f754b3b9d4d5a",
>         	            	    "created": null,
>         	            	    "created_at": "2024-10-15T09:49:03.763Z",
>         	            	    "crn": "crn:v1:bluemix:public:atracker:eu-es:a/abac0df06b644a9cabc6e44f55b3880e::target:6eb48ed0-f6ed-4292-8166-116736fc1242",
>         	            	    "encrypt_key": null,
>         	            	    "encryption_key": null,
>         	            	    "eventstreams_endpoint": "SECURE_VALUE_HIDDEN_HASH:-fb2c5d5039ba8483e4156a315045022aec79794624248cd1d75564a7",
>         	            	    "id": "6eb48ed0-f6ed-4292-8166-116736fc1242",
>         	            	    "logdna_endpoint": "SECURE_VALUE_HIDDEN_HASH:-4d738791de51d8dfb6a23d14aedc12f97d05f1b16c7b5931c1ccd1e8",
>         	            	    "name": "obs-upg-atdyig-es-target",
>         	            	    "region": "eu-es",
>         	            	    "target_type": "event_streams",
>         	            	    "updated": null,
>         	            	    "updated_at": "2024-10-15T09:49:03.763Z",
>         	            	    "write_status": "SECURE_VALUE_HIDDEN_HASH:-36c43efeadd458cc3c1611824565a99f6a7f8d88459ac07b9eb134dc"
>         	            	  },
>         	            	  "after_sensitive": {
>         	            	    "cloudlogs_endpoint": "SECURE_VALUE_HIDDEN_HASH:-ef7222dcb1c2057135074c8d7bdc50670940bf63054659d67ef75f83",
>         	            	    "cos_endpoint": "SECURE_VALUE_HIDDEN_HASH:-1f9ee1a63d4897ef7d6e6a8f4247cd08dd3f2f696a83d897a7c60bac",
>         	            	    "cos_write_status": "SECURE_VALUE_HIDDEN_HASH:-e282313500f3256d1cef6110112ba1612f8a5d69b5aff493ff75d575",
>         	            	    "eventstreams_endpoint": "SECURE_VALUE_HIDDEN_HASH:-84359a6d947319792a2edab3c234cb79a738efa35ce99bca1990942a",
>         	            	    "logdna_endpoint": "SECURE_VALUE_HIDDEN_HASH:-0c5c0023aa42e98e308234f6021ed9a1cd90e11f16367d6f79d308a0",
>         	            	    "write_status": "SECURE_VALUE_HIDDEN_HASH:-045c192f0eb88be23b0ba56e851fe682f9661e8ad6189e8526d6e9de"
>         	            	  },
>         	            	  "after_unknown": {},
>         	            	  "before": {
>         	            	    "api_version": 2,
>         	            	    "cloudlogs_endpoint": "SECURE_VALUE_HIDDEN_HASH:-d1e8947405bf4d1b1cbce22f68b5d9ab879ce3f7b57fad7b5de827db",
>         	            	    "cos_endpoint": "SECURE_VALUE_HIDDEN_HASH:-c44718006a6e82e82691ac6234f7cd88574b88f1939f520a2095214c",
>         	            	    "cos_write_status": "SECURE_VALUE_HIDDEN_HASH:-7a58da3e49702ba6f99e208886389fe88f619d4caf12c33b6fe16ea7",
>         	            	    "created": null,
>         	            	    "created_at": "2024-10-15T09:49:03.763Z",
>         	            	    "crn": "crn:v1:bluemix:public:atracker:eu-es:a/abac0df06b644a9cabc6e44f55b3880e::target:6eb48ed0-f6ed-4292-8166-116736fc1242",
>         	            	    "encrypt_key": null,
>         	            	    "encryption_key": null,
>         	            	    "eventstreams_endpoint": "SECURE_VALUE_HIDDEN_HASH:-2c867119b951db1e9ce1565a49386652b104c820b38233aeaea6872d",
>         	            	    "id": "6eb48ed0-f6ed-4292-8166-116736fc1242",
>         	            	    "logdna_endpoint": "SECURE_VALUE_HIDDEN_HASH:-b7dd0e631d93673ef67b87b19e631df9818a10f716c556d689948fe9",
>         	            	    "name": "obs-upg-atdyig-es-target",
>         	            	    "region": "eu-es",
>         	            	    "target_type": "event_streams",
>         	            	    "updated": null,
>         	            	    "updated_at": "2024-10-15T09:49:03.763Z",
>         	            	    "write_status": "SECURE_VALUE_HIDDEN_HASH:-65104141a864f89fe3c3c7ead660e3ae752044ff5e501d5f617da296"
>         	            	  },
>         	            	  "before_sensitive": {
>         	            	    "cloudlogs_endpoint": "SECURE_VALUE_HIDDEN_HASH:-0fb975d56c5d02406336deef828ee3a0e2e63e7acf1e5868aed58acc",
>         	            	    "cos_endpoint": "SECURE_VALUE_HIDDEN_HASH:-00d224040012a6fe555dda9fcd386b62c4997fee305c2c5e5a04ed4e",
>         	            	    "cos_write_status": "SECURE_VALUE_HIDDEN_HASH:-89fd08bbfa60e19f495a57e9f2407b60600fd45322f0394eadadcd76",
>         	            	    "eventstreams_endpoint": "SECURE_VALUE_HIDDEN_HASH:-4a2c5719ad8fa501afa5c10389b8fcc3edd15606f8d9c0992e9b6661",
>         	            	    "logdna_endpoint": "SECURE_VALUE_HIDDEN_HASH:-895871b303d044568e03ced4d2f61d14b428beeea8a729d1e3408488",
>         	            	    "write_status": "SECURE_VALUE_HIDDEN_HASH:-bc5184fddba5beab98c0b1e172c9e59675aee44a01b031bd4200b145"
>         	            	  }
>         	            	}

File: variables.tf [hunk 1]
```diff
@@ -111,12 +111,13 @@ variable "at_cos_targets" {
...
-    brokers       = list(string)
-    topic         = string
-    api_key       = string # pragma: allowlist secret
-    target_region = optional(string)
-    target_name   = string
+    instance_id                = string
+    brokers                    = list(string)
+    topic                      = string
+    api_key                    = optional(string)
+    service_to_service_enabled = optional(bool, false)
```
Comment (ocofaigh) [inline variables.tf:118]: this should default to true. Your also missing the `skip_atracker_es_iam_auth_policy` option from here (you have it in the submodule)

File: modules/activity_tracker/main.tf [hunk 2]
```diff
@@ -32,6 +32,21 @@ resource "ibm_iam_authorization_policy" "atracker_cloud_logs" {
   description                 = "Permit AT service Sender access to Cloud Logs instance ${each.value.instance_id}"
 }
 
+resource "time_sleep" "wait_for_event_stream_auth_policy" {
```
Comment (ocofaigh) [inline modules/activity_tracker/main.tf:35]: your not using this time_sleep anywhere. It needs to be used in the depends_on for the `ibm_atracker_target.atracker_eventstreams_targets` or else its pointless.

Please also leave a code comment explaining why its used:
`# workaround for https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4478`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
Left some comments

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**